drivers/rtc/rtc_ifx_cat1.c

```diff
@@ -0,0 +1,338 @@
+/*
+ * Copyright (c) 2024 Cypress Semiconductor Corporation (an Infineon company) or
+ * an affiliate of Cypress Semiconductor Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief RTC driver for Infineon CAT1 MCU family.
+ */
+
+#include <zephyr/drivers/rtc.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/device.h>
+#include <cy_pdl.h>
+
+#define DT_DRV_COMPAT infineon_cat1_rtc
+
+#define _IFX_CAT1_RTC_STATE_UNINITIALIZED 0
+#define _IFX_CAT1_RTC_STATE_ENABLED       1
+#define _IFX_CAT1_RTC_STATE_TIME_SET      2
+
+#define _IFX_CAT1_RTC_INIT_CENTURY 2000
+#define _IFX_CAT1_RTC_TM_YEAR_BASE 1900
+
+#if defined(CONFIG_SOC_FAMILY_INFINEON_CAT1B)
+#if defined(SRSS_BACKUP_NUM_BREG3) && (SRSS_BACKUP_NUM_BREG3 > 0)
```

Comment:
Currently Infineon CAT1 integration supports  CAT1A (PSoC 6) and CAT1B (20829). Hare we missed configuration for  CAT1A, please add:
```
#if defined(CONFIG_SOC_FAMILY_INFINEON_CAT1A) 
    #define _IFX_CAT1_RTC_BREG (BACKUP->BREG[SRSS_BACKUP_NUM_BREG-1])
#elif defined(CONFIG_SOC_FAMILY_INFINEON_CAT1C)
   .........
#endif 
   
```

Reply:
For now. This driver is only meant to be set up for the cat1b device.